contrib/golang/common/go/api/filter.go

```diff
@@ -74,9 +74,14 @@ type StreamFilter interface {
 	StreamDecoderFilter
 	// response stream
 	StreamEncoderFilter
+	// log when the request is finished
+	OnLog()
 	// destroy filter
 	OnDestroy(DestroyReason)
-	// TODO add more for stream complete and log phase
+	// TODO add more for stream complete
+}
+
+func (*PassThroughStreamFilter) OnLog() {
```

Comment:
I think a log type parameter should be added, otherwise the API will not know what stage of execution is for Go plugin users.

Reply:
Yes, I am going to pass AccessLogType & allow logging in the middle in a new PR. Currently, the OnLog method is alignment with the Wasm version, which is only called at the end of a stream.